extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageStreamReader.java

```diff
@@ -215,6 +215,7 @@ public BarrageMessage safelyParseFrom(final StreamReaderOptions options,
                     // add and mod rows are never combined in a batch. all added rows must be received before the first
                     // mod rows will be received.
                     if (numAddRowsRead < numAddRowsTotal) {
+                        final long currentChunkFirstRowOffset = lastAddStartIndex;
```

Comment:
We talked about this. We want to dump the globals and start using chunk size/capacity.